### ISSUE TITLE
When I train text with RNN or LSTM, the loss does not decrease.

### ISSUE THREAD

**Issue by Namdosa** [AlexeyAB/darknet#8276]
Hi, all
When I train text with RNN or LSTM, **the loss(avgLoss) does not decrease.**
**Instead, the loss is increasing.** What could be causing the problem?
Thanks.

- OS: Windows10 x64
- GPU:NVIDIA GeForce RTX 2060
- CUDA:cuda_11.5
- CUDANN:8.2.2

- CMD: darknet.exe rnn train cfg/lstm.train.cfg -file t8.shakespeare.txt

 CUDA-version: 11050 (11050), cuDNN: 8.2.2, GPU count: 1
 OpenCV version: 4.5.3
lstm
 0 : compute_capability = 750, cudnn_half = 0, GPU: NVIDIA GeForce RTX 2060
net.optimized_memory = 0
mini_batch = 9216, batch = 73728, time_steps = 576, train = 1
   layer   filters  size/strd(dil)      input                output
   0 LSTM Layer: 256 inputs, 1024 outputs
                Create CUDA-stream - 0
 Create cudnn-handle 0
connected                             256  ->  1024
                connected                             256  ->  1024
                connected                             256  ->  1024
                connected                             256  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
   1 LSTM Layer: 1024 inputs, 1024 outputs
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
   2 LSTM Layer: 1024 inputs, 1024 outputs
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
   3 connected                            1024  ->   256
   4 softmax                                         256
   5 cost                                            256
Total BFLOPS 0.000
avg_outputs = 0
 Allocate additional workspace_size = 4.20 MB
Learning Rate: 0.5, Momentum: 0.9, Decay: 0.001

 batch = 9216, steps = 576, streams = 16, subdivisions = 8, text_size = 5458199
 global_batch = 73728
1: 3.274577, 3.274577 avg, 0.000000 rate, 5.770000 seconds, 0.000000 epochs
2: 3.276542, 3.274774 avg, 0.000000 rate, 3.544000 seconds, 0.000000 epochs
..
341: 3.275620, 3.274915 avg, 0.000000 rate, 3.595000 seconds, 0.000000 epochs
..
429: 3.274759, 3.275004 avg, 0.000050 rate, 3.654000 seconds, 0.001688 epochs
..

- **I used the lstm.train.cfg file as below without any modification.**

[net]
subdivisions=8
inputs=256
batch = 128
momentum=0.9
decay=0.001
max_batches = 2000
time_steps=576
learning_rate=0.5
policy=steps
burn_in=10
steps=1000,1500
scales=.1,.1

[lstm]
batch_normalize=1
output = 1024

[lstm]
batch_normalize=1
output = 1024

[lstm]
batch_normalize=1
output = 1024

[connected]
output=256
activation=leaky

[softmax]

[cost]
type=sse

**Comment by cediy2088** [comment]
same problem

**Comment by cediy2088** [comment]
loss always is 3.274759 ,why?
